src/core/ddsi/include/dds/ddsi/ddsi_tran.h

```diff
@@ -26,6 +26,12 @@
 
 #define DDSI_TRAN_ON_CONNECT 0x0001
 
+#if DDSRT_HAVE_IPV6 == 1
+# define DDSI_LOCATORSTRLEN INET6_ADDRSTRLEN_EXTENDED
```

Comment:
It should be the maximum of all supported address families (which currently can include Ethernet as well). That's not a new issue and it isn't entirely trivial to get the correct value either, so I'm willing to let it be like this for now.